allensdk/test/api/cloud_cache/test_cache.py

```diff
@@ -515,26 +522,37 @@ def test_download_metadata(tmpdir):
 
     cache.load_manifest('manifest_1.json')
 
-    expected_path = cache_dir / true_checksum / 'metadata_file.csv'
+    expected_path = cache_dir / "project4-1" / 'metadata_file.csv'
     assert not expected_path.exists()
 
     # test that metadata_path also works
     attr = cache.metadata_path('metadata_file.csv')
     assert attr['local_path'] == expected_path
     assert not attr['exists']
 
-    result_path = cache.download_metadata('metadata_file.csv')
-    assert result_path == expected_path
-    assert expected_path.exists()
-    hasher = hashlib.blake2b()
-    with open(expected_path, 'rb') as in_file:
-        hasher.update(in_file.read())
-    assert hasher.hexdigest() == true_checksum
+    def response_fun(Bucket, Prefix):
+        # moto doesn't cover list_object_versions
+        return {"Versions": [{
+            "VersionId": meta_version,
+            "Key": "metadata_file.csv",
+            "Size": 12}]}
+    # cache.s3_client.list_object_versions = response_fun
 
-    # test that metadata_path detects that the file now exists
-    attr = cache.metadata_path('metadata_file.csv')
-    assert attr['local_path'] == expected_path
-    assert attr['exists']
+    # NOTE: commenting out because moto does not support
```

Comment:
Err what is meant by `moto does not support list_object_versions`?

You can totally get different versions of objects with a moto bucket. There is just not a straightforward way to lock down the `VersionId` that is returned, but you can still get around that with tests like:

https://github.com/AllenInstitute/informatics_data_release_tools/blob/779178a5712ee1c4cce72c050536b989ed37fe94/tests/remote_clients/test_s3_client.py#L210

Create a ticket to improve quality (and coverage if relevant) of tests but no changes requested.

Reply:
moto does not support the function `list_object_versions`:
![image](https://user-images.githubusercontent.com/32312979/112027360-5523e080-8af4-11eb-87d1-a6dee05edb7c.png)
https://github.com/spulec/moto/blob/master/IMPLEMENTATION_COVERAGE.md